File: core/ui/src/types/material.ts
export type StorageCondition =
  | "AMBIENT"
  | "ROOM_TEMPERATURE"
  | "CONTROLLED_ROOM_TEMPERATURE"
  | "REFRIGERATED_2_TO_8C"
  | "COLD"
  | "DEEP_FREEZER";

export type MaterialCategory =
  | "API"
  | "EXCIPIENT"
  | "SOLVENT"
  | "PACKAGING_MATERIAL"
  | "FINISHED_GOODS"
  | "REFERENCE_STANDARD";

export type LightSensitivity =
  | "NOT_SENSITIVE"
  | "PROTECT_FROM_LIGHT"
  | "AMBER_CONTAINER"
  | "STORE_IN_DARK";

export type Material = {
  id: string;
  materialCode: string;
  materialName: string;
  materialCategory: MaterialCategory | null;
  genericNames: string | null;
  materialType: string;
  uom: string;
  specId: string | null;
  hsnCode: string | null;
  casNumber: string | null;
  pharmacopoeialRef: string | null;
  storageCondition: StorageCondition;
  maxHumidity: string | null;
  lightSensitivity: LightSensitivity | null;
  hygroscopic: boolean;
  shelfLifeMonths: number | null;
  retestPeriodMonths: number | null;
  reorderLevel: string | null;
  leadTimeDays: number | null;
  controlledSubstance: boolean;
  photosensitive: boolean;
  hazardous: boolean;
  selectiveMaterial: boolean;
  vendorCoaReleaseAllowed: boolean;
  samplingRequired: boolean;
  description: string | null;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedBy: string | null;
  updatedAt: string | null;
};

export type CreateMaterialRequest = {
  materialCode: string;
  materialName: string;
  materialCategory?: MaterialCategory;
  genericNames?: string;
  materialType: string;
  uom: string;
  specId: string;
  hsnCode?: string;
  casNumber?: string;
  pharmacopoeialRef?: string;
  storageCondition: StorageCondition;
  maxHumidity?: string;
  lightSensitivity?: LightSensitivity;
  hygroscopic: boolean;
  shelfLifeMonths?: number;
  retestPeriodMonths?: number;
  reorderLevel?: string;
  leadTimeDays?: number;
  controlledSubstance: boolean;
  photosensitive: boolean;
  hazardous: boolean;
  selectiveMaterial: boolean;
  vendorCoaReleaseAllowed: boolean;
  samplingRequired: boolean;
  description?: string;
  createdBy: string;
};

File: core/src/main/java/com/batchsphere/core/masterdata/material/entity/Material.java
package com.batchsphere.core.masterdata.material.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "material")
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Material {

    @Id
    @Column(nullable = false, updatable = false)
    private UUID id;

    @Column(name = "material_code", nullable = false, unique = true, length = 50)
    private String materialCode;

    @Column(name = "material_name", nullable = false, length = 255)
    private String materialName;

    @Column(name = "material_category", length = 50)
    private String materialCategory;

    @Column(name = "generic_names", length = 500)
    private String genericNames;

    @Column(name = "material_type", nullable = false, length = 50)
    private String materialType;

    @Column(name = "uom", nullable = false, length = 50)
    private String uom;

    @Column(name = "spec_id")
    private UUID specId;

    @Enumerated(EnumType.STRING)
    @Column(name = "storage_condition", nullable = false, length = 50)
    private StorageCondition storageCondition;

    @Column(name = "max_humidity", length = 100)
    private String maxHumidity;

    @Column(name = "light_sensitivity", length = 50)
    private String lightSensitivity;

    @Column(name = "photosensitive", nullable = false)
    private Boolean photosensitive;

    @Column(name = "hygroscopic", nullable = false)
    private Boolean hygroscopic;

    @Column(name = "hazardous", nullable = false)
    private Boolean hazardous;

    @Column(name = "selective_material", nullable = false)
    private Boolean selectiveMaterial;

    @Column(name = "vendor_coa_release_allowed", nullable = false)
    private Boolean vendorCoaReleaseAllowed;

    @Builder.Default
    @Column(name = "controlled_substance", nullable = false)
    private Boolean controlledSubstance = false;

    @Column(name = "sampling_required", nullable = false)
    private Boolean samplingRequired;

    @Column(name = "shelf_life_months")
    private Integer shelfLifeMonths;

    @Column(name = "retest_period_months")
    private Integer retestPeriodMonths;

    @Column(name = "reorder_level", length = 100)
    private String reorderLevel;

    @Column(name = "lead_time_days")
    private Integer leadTimeDays;

    @Column(name = "hsn_code", length = 50)
    private String hsnCode;

    @Column(name = "cas_number", length = 100)
    private String casNumber;

    @Column(name = "pharmacopoeial_ref", length = 100)
    private String pharmacopoeialRef;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column(name = "created_by", nullable = false, length = 100)
    private String createdBy;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_by", length = 100)
    private String updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
